pkg/pipeline/ingest/ingest_collector_test.go

```diff
@@ -0,0 +1,54 @@
+package ingest
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/netobserv/flowlogs-pipeline/pkg/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const timeout = 5 * time.Second
+
+func TestIngest(t *testing.T) {
+	collectorPort, err := test.UDPPort()
+	require.NoError(t, err)
+	ic := &ingestCollector{
+		hostname:       "0.0.0.0",
+		port:           collectorPort,
+		batchFlushTime: 10 * time.Millisecond,
+		exitChan:       make(chan bool),
+	}
+	forwarded := make(chan []interface{})
+	defer close(forwarded)
+
+	// GIVEN an IPFIX collector Ingester
+	go ic.Ingest(forwarded)
+
+	client, err := test.NewIPFIXClient(collectorPort)
+	require.NoError(t, err)
+
+	// The IPFIX client might send information before the Ingester is actually listening,
+	// so we might need to repeat the submission
+	test.Eventually(t, timeout, func(t require.TestingT) {
```

Comment:
I really do not like "inconsistent" tests ... can we make this test more robust and drop the Eventually concept ... the entire Eventually concept feels like hiding from creating a consistent and deterministic test.

Reply:
In this case, the "Eventually" is not hiding an inconsistent test but addressing a synchronization issue. Goflow2 is initializing in an internal, background goroutine, and there is no way that we can know when it has started to accept flows (as UDP connections are uni-directional).

There are two ways to address this:
1. Adding a Timeout of e.g. 1-2 seconds, but that would cause that this test is slow.
2. Keep sending flows until we observe that they are being forwarded.

In the next commit, I have rewritten the test to be more explicit that we follow case 2, without giving the impression of flakyness.